dxr/plugins/descriptor/__init__.py

```diff
@@ -0,0 +1,110 @@
+"""Descriptor - Provide useful descriptions for common file types.
+Refer to https://mxr.mozilla.org/webtools-central/source/mxr/Local.pm#27
+"""
+import re
+from os import listdir
+from os.path import splitext, basename, join, isfile
+from warnings import warn
+
+import dxr.indexers
+
+
+class FolderToIndex(dxr.indexers.FolderToIndex):
+    browse_headers = ['Description']
+
+    def needles(self):
+        """If the folder contains a readme, then yield the first line of the
+        readme as the description.
+        Similar to https://mxr.mozilla.org/webtools-central/source/mxr/Local.pm#251
+        """
+        for entry in listdir(self.path):
```

Comment:
We should sort this, as the docs say it comes back in arbitrary order, and it'd be a shame to be nondeterministic.